### PR TITLE
address some browser compatbility issue: replace Map with object lite…

### PR DESCRIPTION
**Pull request by QingpingMeng**
…ral notation,  using setAttribute to set input type to date/time, otherwise IE will throw exception

### CODE REVIEW

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 1]
```diff
@@ -1562,10 +1562,10 @@ export class NumberInput extends Input {
 
     protected internalRender(): HTMLElement {
         this._numberInputElement = document.createElement("input");
-        this._numberInputElement.type = "number";
+        this._numberInputElement.setAttribute('type',"number");
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Quote consistency. Using single and double on the same line
Space after comma

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 2]
```diff
@@ -1562,10 +1562,10 @@ export class NumberInput extends Input {
...
         this._numberInputElement.className = "ac-input ac-numberInput";
-        this._numberInputElement.min = this.min;
-        this._numberInputElement.max = this.max;
+        this._numberInputElement.setAttribute('min', this.min);
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Double quotes for consistency

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 3]
```diff
@@ -1562,10 +1562,10 @@ export class NumberInput extends Input {
...
+        this._numberInputElement.setAttribute('max', this.max);
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Double quotes for consistency

File: source/nodejs/adaptivecards/src/card-elements.ts (hunk 1 above)
Comment (dclaux) [reply source/nodejs/adaptivecards/src/card-elements.ts]: BTW is that setAttribute also something incompatible with a particular browser?

File: source/nodejs/adaptivecards/src/host-config.ts [hunk 4]
```diff
@@ -244,29 +244,28 @@ class BuiltInContainerStyleDefinition extends ContainerStyleDefinition {
...
 
                         if (styleName && typeof styleName === "string") {
-                            if (this._allStyles.has(styleName)) {
-                                this._allStyles.get(styleName).parse(customStyle["style"]);
-                            }
-                            else {
-                                this._allStyles.set(styleName, new ContainerStyleDefinition(customStyle["style"]));
+                            if (this._allStyles.hasOwnProperty(styleName)) {
+                                this._allStyles[styleName].parse(customStyle["style"]);
+                            }else {
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/host-config.ts]: else on new line

File: source/nodejs/adaptivecards/src/host-config.ts [hunk 5]
```diff
@@ -278,17 +277,14 @@ export class ContainerStyleSet {
...
-        this._allStyles.forEach(
-            (value, key, map) => {
-                if (!value.isBuiltIn) {
-                    customStyleArray.push(
-                        {
-                            name: key,
-                            style: value
-                        });
-                }
+        Object.keys(this._allStyles).forEach((key) => {
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/host-config.ts]: Can you put ```(key) => {``` and indent as appropriate on a new line for better legibility?

File: source/nodejs/adaptivecards/src/card-elements.ts (hunk 1 above)
Comment (QingpingMeng) [reply source/nodejs/adaptivecards/src/card-elements.ts]: The problem is setting type to "date" will throw exception in IE causing entire card failed to render but setAttribute will fail silently and keep rendering as a normal input.

Comment (QingpingMeng) [reply source/nodejs/adaptivecards/src/card-elements.ts]: and setAttribute has better compatibility which is even supported by IE8

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 6]
```diff
@@ -1608,7 +1608,7 @@ export class DateInput extends Input {
 
     protected internalRender(): HTMLElement {
         this._dateInputElement = document.createElement("input");
-        this._dateInputElement.setAttribute('type',"date");
+        this._dateInputElement.setAttribute("type","date");
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Missing space after comma

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 7]
```diff
@@ -1633,7 +1633,7 @@ export class TimeInput extends Input {
 
     protected internalRender(): HTMLElement {
         this._timeInputElement = document.createElement("input");
-        this._timeInputElement.setAttribute('type',"time");
+        this._timeInputElement.setAttribute("type","time");
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Missing space after comma

Comment (QingpingMeng) [reply source/nodejs/adaptivecards/src/card-elements.ts]: done

File: source/nodejs/adaptivecards/src/card-elements.ts (hunk 6 above)
Comment (QingpingMeng) [reply source/nodejs/adaptivecards/src/card-elements.ts]: done

**Pull request merged**